### ISSUE TITLE
Question about the loss-function of prompt-based methods.

### ISSUE THREAD

**Issue by PixelChen24** [sun-hailong/LAMDA-PILOT#23]
Hello, I really appreciate your fantastic work!

However, I'm confused about the loss-function of prompt-based methods(l2p, dualprompt, and codaprompt). In their implementations, when calculating the loss after a epoch on a certain task, you mask the "known_classes"(set their logits to be -inf to ignore them in the ce loss)  and calculate loss only on new classes:
```python
logits = output["logits"][:, :self._total_classes]
logits[:, :self._known_classes] = float('-inf')
```
This is the same across all prompt-based methods(both in line 176 of the model file).

But things are different for all methods, which calculated the loss across all seen classes.

So, why does this happen? In my opinion, masking the previous known class will help reduce interference with previous task and reduce forgetting. So is it fair to compare these methods together with different loss design pattern?

**Comment by sun-hailong** [comment]
Hi @PixelChen24,
Thank you for your attention to our work.
Our reproduced method is based on the original code and paper. For prompt-based methods, most of them only update the prompts for the current task while ignoring the previously learned prompts. This is why the logits of self._known_classes are set to -inf. Moreover, since this is only a difference in method design, there is no issue of unfair comparison.

**Comment by PixelChen24** [comment]
I see. Thank you for your response.
